Flaky `pkg/rpcserver` test

<!--
Thanks for submitting a bug report. Please fill the template below,
otherwise we will not be able to process this bug report.
-->

#### Summary
<!-- Summarize the problem in a few sentences: -->

Flaky `pkg/rpcserver` test

#### Steps to Reproduce
<!-- How can we reproduce the problem? -->

1. Run CI undefined amount of times

<!--
Please upload relevant configuration (as .txt).
If you use the command "ttn-lw-stack config", you can redact sensitive config.
-->

#### What do you see now?
<!--
Please paste terminal output, upload logs (as .txt) or upload screenshots.
-->

```
--- FAIL: TestNewRPCServer (0.00s)
    --- FAIL: TestNewRPCServer/Stream (0.00s)
        doc.go:66: 
            /home/travis/gopath/src/go.thethings.network/lorawan-stack/pkg/rpcserver/rpcserver_test.go:138
            Expected collection to have length equal to [2], but it's length was [1] instead! contents: [0xab4ad80]
FAIL
FAIL	go.thethings.network/lorawan-stack/pkg/rpcserver	0.016s
```

#### What do you want to see instead?
<!-- Please add some examples or mock-ups if applicable. -->

Successful CI

#### How do you propose to implement this?
<!-- Please think about how this could be fixed. -->

Debug and fix

#### Can you do this yourself and submit a Pull Request?
<!-- You can also @mention experts if you need help with this. -->

@htdvisser wrote most of this, so I nominate him

--- a/pkg/rpcserver/rpcserver_test.go
+++ b/pkg/rpcserver/rpcserver_test.go
@@ -18,6 +18,7 @@ import (
 	"context"
 	"runtime"
 	"testing"
+	"time"
 
 	"github.com/gogo/protobuf/types"
 	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
@@ -107,6 +108,7 @@ func TestNewRPCServer(t *testing.T) {
 		a.So(mock.pushCtx.Value(&mockKey2{}), should.Resemble, "bar")
 
 		runtime.Gosched()
+		time.Sleep(test.Delay)
 
 		a.So(logHandler.entries, should.HaveLength, 1)
 	})
@@ -134,6 +136,7 @@ func TestNewRPCServer(t *testing.T) {
 		a.So(mock.subCtx.Value(&mockKey2{}), should.Resemble, "foo")
 
 		runtime.Gosched()
+		time.Sleep(test.Delay)
 
 		a.So(logHandler.entries, should.HaveLength, 2)
 	})

